drivers/led/led_gpio.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_leds
+
+/**
+ * @file
+ * @brief GPIO driven LEDs
+ */
+
+#include <drivers/led.h>
+#include <drivers/gpio.h>
+#include <device.h>
+#include <zephyr.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_LOG_LEVEL);
+
+#define DEV_CFG(dev)	((const struct led_gpio_config *) ((dev)->config))
+
+struct led_gpio {
+	const struct device *dev;
+	gpio_pin_t pin;
+	gpio_flags_t flags;
+};
+
+struct led_gpio_config {
+	size_t num_leds;
+	const struct led_gpio *led;
+};
+
+static int led_gpio_blink(const struct device *dev, uint32_t led,
+			  uint32_t delay_on, uint32_t delay_off)
+{
+	return -ENOTSUP;
+}
+
+static int led_gpio_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
+{
+
+	const struct led_gpio_config *config = DEV_CFG(dev);
+	const struct led_gpio *led_gpio;
+
+	if (led >= config->num_leds) {
+		return -EINVAL;
+	}
+
+	if ((value != 100) && (value != 0)) {
```

Comment:
can you please do it as
`value<50` then off
`value>=50` then on

this way any client using brightness will work without having to consider that this driver behaves differently.

Reply:
Fixed